backend/src/main/java/com/redhat/cloud/notifications/routers/IncomingRequestInterceptor.java

```diff
@@ -0,0 +1,92 @@
+
+@Provider
+@PreMatching
+public class IncomingRequestInterceptor implements ContainerRequestFilter {
+
+
+    // Prevents the injection of characters that would break the log file pattern and lead to log forging or log poisoning.
+    private static final Pattern ANTI_INJECTION_PATTERN = Pattern.compile("[\n|\r|\t]");
+
+    private static final Pattern p = Pattern.compile("/api/(integrations|notifications)/v(\\d+)/(.*)");
+
+    @Override
+    public void filter(ContainerRequestContext requestContext) throws IOException {
+        openapiAcceptHeaderMangler(requestContext);
+
+        routeRedirector(requestContext);
+
+        addOrgIdToMDCForAccessLogs(requestContext);
+    }
+
+    /**
+     * Change the accept header if needed for Openapi requests when
+     * openapi.json is requested.
+     */
+    private static void openapiAcceptHeaderMangler(ContainerRequestContext requestContext) {
+        /*
+         * CPOL-107
+         * Default return format for openapi is .yml
+         * If the user requests 'openapi.json', the user assumes
+         * that a JSON format is returned. Unfortunately does Quarkus not
+         * honor the '.json' suffix but either requires a correct Accept
+         * header or the use of a query parameter.
+         *
+         * We now look at the path and if it ends in .json, replace the
+         * existing Accept header with one that requests Json format.
+         */
+
+        if (requestContext.getUriInfo().getPath().endsWith("openapi.json")) {
+            requestContext.getHeaders().remove("Accept");
+            requestContext.getHeaders().add("Accept", "application/json");
+        }
+    }
+
+    /**
+     * Add org id to Mapped diagnostic context to be accessible from accessLogs
+     */
+    private static void addOrgIdToMDCForAccessLogs(ContainerRequestContext requestContext) {
+        if (requestContext.getHeaders().containsKey(X_RH_IDENTITY_HEADER)) {
+            ConsoleIdentity identity = ConsoleIdentityProvider.getRhIdentityFromString(requestContext.getHeaders().getFirst(X_RH_IDENTITY_HEADER));
+            if (identity instanceof RhIdentity rhIdentity) {
+                MDC.put("x-rh-org-id", rhIdentity.getOrgId());
+            }
+        }
+    }
+
+    /**
+     * If the requested route is the one with major version only,
+     * we rewrite it on the fly.
+     */
+    private void routeRedirector(ContainerRequestContext requestContext) {
+
+        String uri = requestContext.getUriInfo().getPath();
+        if (Log.isTraceEnabled()) {
+            String sanitizedUri = ANTI_INJECTION_PATTERN.matcher(uri).replaceAll("");
+            Log.tracef("Incoming uri: %s", sanitizedUri);
+        }
+        Matcher m = p.matcher(uri);
+        if (m.matches()) {
+            String newTarget = "/api/" + m.group(1) + "/v" + m.group(2) + ".0/" + m.group(3);
+            Log.tracef("Rerouting uri path: %s to %s", requestContext.getUriInfo().getPath(), newTarget);
```

Comment:
This line :point_up: is vulnerable to log forging/poisoning because the URI path isn't sanitized. I would stick with the old log message from `RouteRedirector` because we're already logging the initial incoming URI at line 82 in this code.
```suggestion
            Log.tracef("Rerouting to: %s", newTarget);
```